Simplicity/LoginProviders/VKontakte.swift

```diff
@@ -33,7 +33,7 @@ public class VKontakte: OAuth2 {
             let range = urlScheme.range(of: "\\d+", options: .regularExpression) else {
                 preconditionFailure("You must configure your VK URL Scheme to use VK login.")
         }
-        let clientId = urlScheme.substring(with: range)
+        let clientId = String(urlScheme[range])
         let authorizationEndpoint = URL(string: "https://oauth.vk.com/authorize")!
```

Comment:
do you have error about undeclared URL? because Foundation is not imported here.